src/sas/sascalc/dataloader/readers/IgorReader.py

```diff
@@ -39,223 +38,157 @@ class Reader:
     def read(self, filename=None):
         """ Read file """
         if not os.path.isfile(filename):
-            raise ValueError, \
-            "Specified file %s is not a regular file" % filename
-        
-        # Read file
-        f = open(filename, 'r')
-        buf = f.read()
+            raise ValueError("Specified file %s is not a regular "
+                             "file" % filename)
         
-        # Instantiate data object
         output = Data2D()
+
         output.filename = os.path.basename(filename)
         detector = Detector()
-        if len(output.detector) > 0:
-            print str(output.detector[0])
+        if len(output.detector):
+            print(str(output.detector[0]))

```

Comment:
I have no idea why the print statement is there. The detector object has just been created, so I don't see the point in printing it's length (if it has one).

I'll make that ticket.